src/cswinrt/code_writers.h

```diff
@@ -1805,87 +1808,89 @@ remove => %;
-    void write_abi_method_with_raw_return_type(writer& w, MethodDef const& method);
+        auto args = attribute.Value().FixedArgs();
 
-    template<auto method_writer>
-    std::string write_factory_cache_object(writer& w, TypeDef const& factory_type, TypeDef const& class_type);
+        using std::get;
+
+        auto get_arg = [&](decltype(args)::size_type index) { return get<ElemSig>(args[index].value).value; };
 
-    std::string write_static_cache_object(writer& w, std::string_view cache_type_name, TypeDef const& class_type)
+        w.write_printf(R"(0x%X, 0x%X, 0x%X, 0x%X, 0x%X, 0x%X, 0x%X, 0x%X, 0x%X, 0x%X, 0x%X, 0x%X, 0x%X, 0x%X, 0x%X, 0x%X)",
+            (get<uint32_t>(get_arg(0)) >> 0) & 0xFF,
+            (get<uint32_t>(get_arg(0)) >> 8) & 0xFF,
+            (get<uint32_t>(get_arg(0)) >> 16) & 0xFF,
+            (get<uint32_t>(get_arg(0)) >> 24) & 0xFF,
+            (get<uint16_t>(get_arg(1)) >> 0) & 0xFF,
+            (get<uint16_t>(get_arg(1)) >> 8) & 0xFF,
+            (get<uint16_t>(get_arg(2)) >> 0) & 0xFF,
+            (get<uint16_t>(get_arg(2)) >> 8) & 0xFF,
+            get<uint8_t>(get_arg(3)),
+            get<uint8_t>(get_arg(4)),
+            get<uint8_t>(get_arg(5)),
+            get<uint8_t>(get_arg(6)),
+            get<uint8_t>(get_arg(7)),
+            get<uint8_t>(get_arg(8)),
+            get<uint8_t>(get_arg(9)),
+            get<uint8_t>(get_arg(10)));
+    }
+
+    void write_class_static_cache_definition(writer& w, TypeDef const& staticsType, TypeDef const& classType)
     {
-        auto instance =
-            w.write_temp(
-                "private static readonly _% _instance = new _%();",
-                cache_type_name,
-                cache_type_name);
-        auto factoryAs = w.write_temp("%",
-            bind([&](writer& w)
-            {
-                if (is_static(class_type))
-                {
-                    w.write("_factory._As");
-                }
-                else
-                {
-                    w.write("ActivationFactory<%>.As", class_type.TypeName());
-                }
-            })
-        );
+        if (staticsType)
+        {
+            auto factory_class_name = settings.netstandard_compat ?
+                w.write_temp("BaseFactory<%.Vftbl>", bind<write_type_name>(staticsType, typedef_name_type::ABI, true)) :
+                w.write_temp("BaseFactory");
 
-        if (settings.netstandard_compat)
-        {            
-        auto cache_vftbl_type = w.write_temp("ABI.%.%.Vftbl",
-                class_type.TypeNamespace(),
-                cache_type_name);
-        auto cache_interface =
-            w.write_temp(
-                R"(%<%>)",
-                factoryAs,
-                cache_vftbl_type);
+            auto statics_type_name = staticsType.TypeName();
             w.write(R"(
-internal sealed class _% : ABI.%.%
-{
-public _%() : base(%()) { }
-%
-internal static % Instance => _instance;
-}
+private static % _% = new %("%.%", %.IID);
```

Comment:
Is this ever reassigned at runtime? If not, shouldn't this be `readonly` to improve codegen?